sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/plan/PlanCustomizer.java

```diff
@@ -4,5 +4,22 @@
  * This interface allows an opportunity for plans to be modified before execution.
  */
 public interface PlanCustomizer {
+
+    /**
+     * Takes a reference to a plan, returning a reference to a modified plan derived from the original.
+     * @param plan
+     * @return
+     */
     Plan updatePlan(Plan plan);
```

Comment:
Could have a default no-op here too, if e.g. someone only wanted to customize the uninstall side.